docs: document migratable RWX volumes to distinguish from standard RWX volumes (#1209)
longhorn/website: content/docs/1.11.0/nodes-and-volumes/volumes/rwx-volumes.md

Longhorn supports ReadWriteMany (RWX) volumes by exposing regular Longhorn volumes via NFSv4 servers that reside in share-manager pods.

## Introduction

Longhorn provides two types of RWX volumes, each optimized for different workload requirements.

> - **Standard (Non-Migratable) RWX Volumes**

Reviewer: ```suggestion
- **Generic (Non-Migratable) RWX Volumes**
```

  Standard RWX volumes provide traditional shared filesystem access across multiple nodes. They use dedicated NFSv4.1 servers running in `share-manager-<volume-name>` Pods within the longhorn-system namespace. Each RWX volume is paired with a corresponding Service that exposes the NFS endpoint to clients.

  These volumes are ideal for workloads that need concurrent file access but do not require live migration.

  Characteristics:

>   - Not capable of live migration

Reviewer: ```suggestion
  - Not capable of [live migration](https://kubevirt.io/user-guide/compute/live_migration/)
```

# Introduction

Longhorn supports two types of ReadWriteMany (RWX) volumes:

> 1. **Standard (Non-Migratable) RWX Volumes**: Traditional shared storage using NFSv4 servers in dedicated `share-manager-<volume-name>` Pods within the `longhorn-system` namespace. These volumes provide shared access across multiple nodes but cannot be live migrated. Standard RWX volumes create a corresponding Service for each RWX volume, serving as the designated endpoint for NFSv4 client connections.

Reviewer 2: @copilot Should be NFSv4.1

**Use Cases:**
- Traditional shared storage scenarios
- Applications that require concurrent read/write access from multiple nodes
- Workloads that don't require live migration capabilities

**Configuration Example:**
```yaml
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: longhorn-rwx-standard
provisioner: driver.longhorn.io
allowVolumeExpansion: true
reclaimPolicy: Delete
volumeBindingMode: Immediate
parameters:
  numberOfReplicas: "3"
  staleReplicaTimeout: "2880"
  migratable: "false"  # Explicitly disabled for standard RWX
```

## Migratable RWX Volumes

**Migratable RWX volumes** support live migration from one node to another, making them ideal for virtualized workloads like KubeVirt VMs that require live migration capabilities.

> **Notice**  
> Migratable RWX volumes must have the access mode set to `ReadWriteMany`, the "migratable" flag enabled (*parameters.migratable: `true`*), and use `volumeMode: Block`.

**Use Cases:**
- Virtual machines requiring live migration
- Workloads that benefit from live migration for maintenance or load balancing

**Key Features:**
- Supports live migration between nodes

> - Uses block device mode (`volumeMode: Block`)
> - Manages dual engine instances during migration
> - Automatic migration confirmation or rollback based on pod lifecycle
> 
> **How Migration Works:**
> 1. **Migration Start**: When Kubernetes requests attachment of an already-attached migratable volume to a new node, Longhorn creates a second engine on the destination node with matching replicas
> 2. **Dual Engine State**: Both engines run simultaneously, with the volume accessible from both nodes during the migration period
> 3. **Migration Completion**: Based on which pod is terminated:
>    - If the original pod is terminated → migration completes, volume moves to the new node
>    - If the new pod is terminated → migration rolls back, volume remains on the original node
> 4. **Cleanup**: Longhorn automatically cleans up the unused engine and adjusts replica states

Reviewer 3: ```suggestion
- Utilizes block device mode (`volumeMode: Block`) for efficient data synchronization
- Manages dual engine instances during the migration process
- Automatic completion or rollback based on the application's state transition

**How Migration Works:**
1. **Orchestration Request**: A live migration is initiated when a pod or virtual machine is rescheduled to a new node, causing Kubernetes to request volume attachment on the destination.
2. **Dual Engine State**: Longhorn creates a second engine instance on the new node. Both the original and new engines run simultaneously, with the volume accessible from both nodes. Data is continuously synchronized at the block level.
3. **Completion/Rollback**: The migration's final state depends on the application's lifecycle:
   - **Completion**: When the application successfully starts on the new node, the original instance is terminated. Longhorn recognizes this as a successful migration.
   - **Rollback**: If the new application instance fails to start or is terminated, the migration is automatically rolled back, and the volume remains attached to the original node.
4. **Cleanup**: After the final state is confirmed, Longhorn cleans up the unused engine instance and updates the volume's replica status.
```

> **Notice**  
> Standard RWX volumes must have the access mode set to `ReadWriteMany` and the "migratable" flag disabled (*parameters.migratable: `false`*).

**Use Cases:**
- Traditional shared storage scenarios
- Applications that require concurrent read/write access from multiple nodes
- Workloads that don't require live migration capabilities

**Configuration Example:**
```yaml
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: longhorn-rwx-standard
provisioner: driver.longhorn.io
allowVolumeExpansion: true
reclaimPolicy: Delete
volumeBindingMode: Immediate
parameters:
  numberOfReplicas: "3"
  staleReplicaTimeout: "2880"
  migratable: "false"  # Explicitly disabled for standard RWX
```

## Migratable RWX Volumes

**Migratable RWX volumes** support live migration from one node to another, making them ideal for virtualized workloads like KubeVirt VMs that require live migration capabilities.

> **Notice**  
> Migratable RWX volumes must have the access mode set to `ReadWriteMany`, the "migratable" flag enabled (*parameters.migratable: `true`*), and use `volumeMode: Block`.

**Use Cases:**

> - KubeVirt virtual machines requiring live migration

Reviewer 2: @copilot Remove `KubeVirt`. `virtual machines` is good enough.

> - Harvester VMs that need to move between nodes without downtime

Reviewer 2: @copilot Remove `harvester`. Replace `VMs` with `Virtual machines` VMs are good enough.

    There is a dedicated recovery backend service for NFS servers in Longhorn system. When a client connects to an NFS server, the client's information, including its hostname, will be stored in the recovery backend. When a share-manager Pod or NFS server is abnormally terminated, Longhorn will create a new one. Within the 90-seconds grace period, clients will reclaim locks using the client information stored in the recovery backend.

> ## Creation and Usage of Standard (Non-Migratable) RWX Volumes

Reviewer: ```suggestion
## Creation and Usage of Generic (Non-Migratable) RWX Volumes
```
